cf-agent/acl_posix.c

```diff
@@ -339,15 +341,33 @@ static int CheckPosixLinuxACEs(EvalContext *ctx, Rlist *aces, AclMethod method,
 
             if (!DONTDO)
             {
+                int last = -1;
+                acl_text_str = acl_to_any_text(acl_new, NULL, ',', 0);
+                Log(LOG_LEVEL_DEBUG, "ACL: new acl is `%s'", acl_text_str);
+
+                if ((retv = acl_check(acl_new, &last)) != 0)
+                {
+                    Log(LOG_LEVEL_ERR, "ACL: acl_check(%s): %s at entry %d. Refer to acl_check man page for validity rules.",
+			            acl_text_str,
+        			    acl_error(retv),
+        			    last);
+                    acl_free((void *) acl_existing);
+                    acl_free((void *) acl_tmp);
+                    acl_free((void *) acl_new);
+         		    acl_free(acl_text_str);
```

Comment:
`acl_free()` is not for strings. Just use `free()`.

Reply:
I took the hint from acl_to_text(3) ```This function allocates any memory necessary to contain the string and returns a pointer to the string.  The caller should free
     any releasable memory, when the new string is no longer required, by calling acl_free(3) with the (void*)char returned by
     acl_to_text() as an argument.
``` Same difference I suppose but it seems more proper to use acl_free()?

Reply:
You are right, ignore this comment then! (copied misplaced comment below)